source/Playlist.cpp

```diff
@@ -0,0 +1,134 @@
+/* Playlist.cpp
+Copyright (c) 2022 by RisingLeaf
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Playlist.h"
+
+#include "DataNode.h"
+#include "GameData.h"
+#include "PlayerInfo.h"
+#include "System.h"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace {
+	const Track *currentTrack = nullptr;
+}
+
+
+
+Playlist::Playlist(const DataNode &node)
+{
+	Load(node);
+}
+
+
+
+void Playlist::Load(const DataNode &node)
+{
+	// All tracks need a name.
+	if(node.Size() < 2)
+	{
+		node.PrintTrace("Error: No name specified for playlist:");
+		return;
+	}
+
+	if(!name.empty())
+	{
+		node.PrintTrace("Error: Duplicate definition of playlist:");
+		return;
+	}
+	name = node.Token(1);
+
+	for(const DataNode &child : node)
+	{
+		if(child.Token(0) == "to")
+		{
+			if(child.Token(1) == "play")
+				toPlay.Load(child);
+		}
+		else if(child.Token(0) == "location")
+			location.Load(child);
+		else if(child.Token(0) == "priority" && child.Size() >= 2)
+			priority = child.Value(1);
+		else if(child.Token(0) == "weight" && child.Size() >= 2)
+			weight = child.Value(1);
+		else if(child.Token(0) == "tracks")
+		{
+			if(child.Size() >= 2)
+				progressionStyle = child.Token(1);
+			else
+				progressionStyle = "linear";
+			for(const auto &grand : child)
+				if(grand.Size() >= 2)
+					tracks.emplace_back(grand.Value(1), GameData::Tracks().Get(grand.Token(0)));
+				else
+					tracks.emplace_back(10, GameData::Tracks().Get(grand.Token(0)));
```

Comment:
The default track weight is `10` if the content creator does not specify a track weight.  I asked RisingLeaf about this.  We may want to change the default weight to `2`.

Also, the game crashes with an exception thrown any time a content creator specifies a `weight` less than `2`.  We should probably throw a syntax warning and ignore the weight value instead of crashing.

Reply:
Fixed; now default is 1